### PR TITLE
Fix FUSE hardlinks

### PR DESCRIPTION
**Pull request by ThomasWaldmann**

### CODE REVIEW

**Comment by ThomasWaldmann** [comment]
merge as is?

File: src/borg/fuse.py [hunk 1]
```diff
@@ -43,7 +43,9 @@ def add(self, item):
         return pos + self.offset
 
     def get(self, inode):
-        self.fd.seek(inode - self.offset, io.SEEK_SET)
+        offset = inode - self.offset
+        assert offset >= 0, 'ItemCache.get() called with a invalid inode number'
```
Comment (enkore) [inline src/borg/fuse.py]: Hm, better use an if...:ValueError?

File: src/borg/fuse.py [hunk 2]
```diff
@@ -133,46 +135,66 @@ def _create_dir(self, parent):
...
         """Build fuse inode hierarchy from archive metadata
         """
+        self.file_versions = {}  # for versions mode: original path -> version
         unpacker = msgpack.Unpacker()
         for key, chunk in zip(archive.metadata.items, self.repository.get_many(archive.metadata.items)):
             _, data = self.key.decrypt(key, chunk)
             unpacker.feed(data)
             for item in unpacker:
                 item = Item(internal_dict=item)
+                path = os.fsencode(os.path.normpath(item.path))
```
Comment (enkore) [inline src/borg/fuse.py]: I see that this came from the old code, but is normpath actually necessary? We normalize paths in archives.

Comment (ThomasWaldmann) [inline src/borg/fuse.py]: maybe not, but i think this should be done seperately from this changeset (and needs a global + old code review).

File: src/borg/fuse.py (hunk 1 above)
Comment (ThomasWaldmann) [inline src/borg/fuse.py]: ok

File: src/borg/fuse.py (hunk 2 above)
Comment (enkore) [inline src/borg/fuse.py]: I'm ok with that :)

**Pull request merged**